### PR TITLE
HPCC4J-581 WsFS Client should add path delim only if needed

### PR DESCRIPTION
**Pull request by rpastrana**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [ ] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [x] The change has been fully tested:
  - [ ] This change does not cause any existing JUnits to fail.
  - [ ] I have include JUnit coverage to test this change
  - [ ] I have performed system test and covered possible regressions and side effects.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] This change fixes the problem, not just the symptom

### CODE REVIEW

**Comment by rpastrana** [comment]
@jpmcmu this is currently targeting 9.4.x, but considering back-porting, unless you can think of a reason why we shouldn't

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCFileSprayClient.java [hunk 1]
```diff
@@ -1004,9 +1004,8 @@ public ProgressResponseWrapper sprayVariable(DelimitedDataOptions options, DropZ
         if (targetDropZone == null) throw new Exception("TargetDropZone object not available!");
 
         SprayVariable request = new SprayVariable();
-
         request.setSourceIP(targetDropZone.getNetAddress());
-        request.setSourcePath(targetDropZone.getPath() + "/" + sourceFileName);
+        request.setSourcePath(Utils.ensureTrailingPathSlash(targetDropZone.getPath()) + sourceFileName);
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCFileSprayClient.java:1008]: Do these calls to ensureTrailingPathSlash also need the call to trim()?

**Review by jpmcmu — APPROVED** [review]
Looks good to me, a minor question about needing trim()

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCFileSprayClient.java (hunk 1 above)
Comment (rpastrana) [reply wsclient/src/main/java/org/hpccsystems/ws/client/HPCCFileSprayClient.java:1008]: it's been the user's responsibility to do so, added a java11 instruction to strip all trailing white space chars

**Comment by rpastrana** [comment]
@jpmcmu please review

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCFileSprayClient.java (hunk 1 above)
Comment (rpastrana) [reply wsclient/src/main/java/org/hpccsystems/ws/client/HPCCFileSprayClient.java:1008]: as discussed offline, we will continue to target java8 for now.
I added a custom trimRight function instead.

**Review by jpmcmu — APPROVED** [review]
@rpastrana Looks good to me

**Review by jpmcmu — APPROVED** [review]
@rpastrana looks good, are you still planning on backporting this to 9.0x?

**Pull request closed without merging**